src/commands/captureDiagnostics.ts

```diff
@@ -0,0 +1,281 @@
+    allowMinimalCapture: boolean = true
+) {
+    try {
+        const captureMode = await captureDiagnosticsMode(ctx, allowMinimalCapture);
+
+        // dialog was cancelled
+        if (!captureMode) {
+            return;
+        }
+
+        const diagnosticsDir = path.join(
+            tmpdir(),
+            `vscode-diagnostics-${formatDateString(new Date())}`
+        );
+
+        await fs.mkdir(diagnosticsDir);
+        await writeLogFile(diagnosticsDir, "extension-logs.txt", extensionLogs(ctx));
+        await writeLogFile(diagnosticsDir, "settings.txt", settingsLogs(ctx));
+        await writeLogFile(diagnosticsDir, "diagnostics.txt", diagnosticLogs());
+
+        if (captureMode === "Full") {
+            // The `sourcekit-lsp diagnose` command is only available in 6.0 and higher.
+            if (ctx.swiftVersion.isGreaterThanOrEqual(new Version(6, 0, 0))) {
+                await sourcekitDiagnose(ctx, diagnosticsDir);
+            } else {
+                await writeLogFile(diagnosticsDir, "sourcekit-lsp.txt", sourceKitLogs(ctx));
+            }
+        }
+
+        ctx.outputChannel.log(`Saved diagnostics to ${diagnosticsDir}`);
+        await showCapturedDiagnosticsResults(diagnosticsDir);
+    } catch (error) {
+        vscode.window.showErrorMessage(`Unable to capture diagnostic logs: ${error}`);
+    }
+}
+
+export async function promptForDiagnostics(ctx: WorkspaceContext) {
+    const ok = "OK";
+    const cancel = "Cancel";
+    const result = await vscode.window.showInformationMessage(
+        "SourceKit-LSP has been restored. Would you like to capture a diagnostic bundle to file an issue?",
+        ok,
+        cancel
+    );
+
+    if (!result || result === cancel) {
+        return;
+    }
+
+    return await captureDiagnostics(ctx, false);
+}
+
+async function captureDiagnosticsMode(
+    ctx: WorkspaceContext,
+    allowMinimalCapture: boolean
+): Promise<"Minimal" | "Full" | undefined> {
+    if (
+        ctx.swiftVersion.isGreaterThanOrEqual(new Version(6, 0, 0)) ||
+        vscode.workspace.getConfiguration("sourcekit-lsp").get<string>("trace.server", "off") !==
+            "off"
+    ) {
+        const fullButton = allowMinimalCapture ? "Capture Full Diagnostics" : "Capture Diagnostics";
+        const minimalButton = "Capture Minimal Diagnostics";
+        const buttons = allowMinimalCapture ? [fullButton, minimalButton] : [fullButton];
+        const fullCaptureResult = await vscode.window.showInformationMessage(
+            `A Diagnostic Bundle collects information that helps the developers of the VS Code Swift extension diagnose and fix issues.
+
+This information contains:
+- Extension logs
+- Versions of Swift installed on your system
+- Crash logs from SourceKit
+- Log messages emitted by SourceKit
+- If possible, a minimized project that caused SourceKit to crash
+- If possible, a minimized project that caused the Swift compiler to crash
+
+Please attach this bundle to GitHub issues.`,
```

Comment:
Should we have a call for action here to file an issue? I think this would be especially valuable when a user captures a diagnostic bundle after sourcekitd has crashed. Something like

```suggestion
Please file an issue with a description of the problem you are seeing and this diagnose bundle at https://github.com/swiftlang/vscode-swift.`,
```

Or use `apple/sourcekit-lsp`, I don’t have strong opinions here. It’ s pretty easy to route between the two.